New validation for CFD CSCwf26626

Harinadh-Saladi opened the pull request:
Added additional test cases and modifed validations.md file

Harinadh-Saladi committed:
Added disabled_cipher_check for CSCwf26626

- Implemented new check function to detect disabled cipher configuration issues
- Added comprehensive test suite with 6 test cases covering all scenarios
- Update validations.md documentation with check details and recommendations

Harinadh-Saladi committed:
Fix fabricNode.json: Replace leaf node with third APIC controller (apic3)

Harinadh-Saladi committed:
Updated test data JSON files to array format with full cipher attributes

Harinadh-Saladi committed:
Added additional test cases for disabled_cipher_check and modified validations.md file as it was having a typo

lovkeshsharma702 reviewed (changes requested):
logic is not correct. it just verify the version and fail, In fab3 apic1, all cph are enabled and no nginx error but the check still fail.  
Please work on correcting, testing, validating the main function and test cases.

Harinadh-Saladi committed:
Fixed nginx log check: Filtered out command echo and prompt correctly

Harinadh-Saladi commented:
Enclosing the pytest logs for the same.
[Pytest logs.txt](https://github.com/user-attachments/files/24235190/Pytest.logs.txt)

Harinadh-Saladi committed:
Updated disabled_cipher_check with nginx log filtering fix

Harinadh-Saladi committed:
Removed break from line number 1608 as it was added to proceed with the script execution for earlier issue as a WA

Harinadh-Saladi committed:
Merging latest master changes

diwakarmishra1190 commented:
Code logic  looks good .
all test passed.

diwakarmishra1190 reviewed (approved):
tested locally .
LGTM.

lovkeshsharma702 commented on aci-preupgrade-validation-script.py:
```diff
@@ -5970,6 +5970,121 @@ def configpush_shard_check(tversion, **kwargs):
 
     return Result(result=result, headers=headers, data=data, recommended_action=recommended_action, doc_url=doc_url)
 
+<<<<<<< HEAD
```
why do we have <<<<<< HEAD ?

lovkeshsharma702 reviewed (changes requested):
requested change.

Harinadh-Saladi committed:
Remove merge conflict markers from previous commit

- Removed <<<<<<< HEAD, =======, and >>>>>>> upstream/master markers
- Kept both disabled_cipher_check and apic_vmm_inventory_sync_faults_check functions
- Fixed improper merge from commit 6cb1209

Harinadh-Saladi committed:
Removed target version missing check since when tversion is not provided it will prompt for tversion to provide due to the recent changes done in the main code

diwakarmishra1190 reviewed (approved):
Approved.

Harinadh-Saladi commented:
Closing the PR as suggested by Lovkesh and Team.
